src/Orleans/Utils/IKeyedServiceCollection.cs

```diff
@@ -0,0 +1,37 @@
+﻿
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Collection of services that can be disambiguated by key
+    /// </summary>
+    public interface IKeyedServiceCollection<TKey, TService>
+        where TService : class
+    {
+        TService GetService(TKey key);
+    }
+
+    public static class KeyedServiceExtensions
+    {
+        /// <summary>
+        /// Acquire a service by key.
+        /// </summary>
+        public static TService GetServiceByKey<TKey, TService>(this IServiceProvider services, TKey key)
```

Comment:
It depends on the nature of the UI piece (window, message/popup box, etc.). But yeah, Ok, Cancel is the "right" order form LTR, but default is not always the Ok. :D